datasets/austin_crime/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,252 @@
+
+
+def main(
+    source_url: typing.List[str],
+    source_file: typing.List[pathlib.Path],
+    source_files_path: str,
+    target_file: pathlib.Path,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+    headers: typing.List[str],
+    rename_mappings: dict,
+) -> None:
+
+    logging.info(
+        "Austin crime process started at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+    logging.info("creating 'files' folder")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+
+    logging.info("Downloading file ...")
+    download_file(source_url, source_file)
+
+    logging.info("Opening files...")
+    df = read_files(source_files_path)
+
+    logging.info("Transform: Rename columns...")
+    rename_headers(df, rename_mappings)
+
+    logging.info("Transform: Cleaning up column location_description...")
+    df["location_description"] = (
+        df["location_description"]
+        .astype("|S")
+        .str.decode("utf-8")
+        .apply(reg_exp_tranformation, args=(r"\s{2,}", ""))
+    )
+
+    logging.info("Transform: Converting to integer string...")
+    df["zipcode"] = df["zipcode"].apply(convert_to_integer_string)
+    df["council_district_code"] = df["council_district_code"].apply(
+        convert_to_integer_string
+    )
+    df["x_coordinate"] = df["x_coordinate"].apply(convert_to_integer_string)
+    df["y_coordinate"] = df["y_coordinate"].apply(convert_to_integer_string)
+
+    logging.info("Transform: Creating a new column - address...")
+    df["address"] = df["address1"]
+    df["address"] = (
+        df["address"]
+        .fillna(
+            df["location_description"].replace("nan", "")
+            + " Austin, TX "
+            + df["zipcode"]
+        )
+        .str.strip()
+    )
+
+    logging.info("Transform: Converting date format...")
+    df["timestamp"] = df["timestamp"].apply(convert_dt_format)
+    df["clearance_date"] = df["clearance_date"].apply(convert_dt_format)
+
+    logging.info("Transform: Creating a new column - year...")
+    df["year"] = df["timestamp"].apply(extract_year)
+
+    logging.info("Transform: Replacing values...")
+    df["address"] = df["address"].apply(reg_exp_tranformation, args=(r"\n", " "))
+    df = df.replace(
+        to_replace={
+            "clearance_status": {
+                "C": "Cleared by Arrest",
+                "O": "Cleared by Exception",
+                "N": "Not cleared",
+            },
+            "address": {"sAustin": "Austin"},
+        }
+    )
+
+    logging.info("Transform: Converting exponential values to integer...")
+    df["unique_key"] = (
+        df["unique_key"]
+        .apply(convert_exp_to_float)
+        .astype(float)
+        .apply(convert_to_integer_string)
+    )
+
+    logging.info("Transform: Creating a new column - latitude...")
+    df["latitude"] = (
+        df["address"]
+        .apply(search_string)
+        .apply(extract_lat_long, args=[r".*\((\d+\.\d+),.*"])
+    )
```

Comment:
Also include a comment with an example of what this column looks like

Reply:
I added comments with example

Reply:
I don't seem to find any examples in the log statement?

Reply:
I have added example as comment